DQM/CTPPS/plugins/CTPPSRandomDQMSource.cc

```diff
@@ -0,0 +1,201 @@
+    RPindexValid[index] = 0;
+}
+
+//-------------------------------------------------------------------------------------
+
+void CTPPSRandomDQMSource::bookHistograms(DQMStore::IBooker &ibooker, edm::Run const &, edm::EventSetup const &) {
+  ibooker.cd();
+  ibooker.setCurrentFolder("CTPPS/TrackingPixel");
+
+  hBX = ibooker.book1D("events per BX", "ctpps_pixel;Event.BX", 4002, -1.5, 4000. + 0.5);
+
+  for (int arm = 0; arm < NArms; arm++) {
+    CTPPSDetId ID(CTPPSDetId::sdTrackingPixel, arm, 0);
+    string sd;
+    ID.armName(sd, CTPPSDetId::nPath);
+
+    ibooker.setCurrentFolder(sd);
+
+    for (int stn = 0; stn < NStationMAX; stn++) {
+      if (StationStatus[stn] == 0)
+        continue;
+      ID.setStation(stn);
+      string stnd;
+      CTPPSDetId(ID.stationId()).stationName(stnd, CTPPSDetId::nPath);
+
+      ibooker.setCurrentFolder(stnd);
+
+      for (int rp = RPn_first; rp < RPn_last; rp++) {  // only installed pixel pots
+        ID.setRP(rp);
+        string rpd, rpTitle;
+        CTPPSDetId(ID.rpId()).rpName(rpTitle, CTPPSDetId::nFull);
+        CTPPSDetId(ID.rpId()).rpName(rpd, CTPPSDetId::nPath);
+
+        ibooker.setCurrentFolder(rpd);
+
+        int indexP = getRPindex(arm, stn, rp);
+        RPindexValid[indexP] = 1;
+
+        h2HitsVsBXRandoms[indexP] = ibooker.book2D("Digi per plane per BX - random triggers",
+                                                   rpTitle + ";Event.BX;Plane",
+                                                   4002,
+                                                   -1.5,
+                                                   4000. + 0.5,
+                                                   NplaneMAX,
+                                                   0,
+                                                   NplaneMAX);
+
+      }  // end for(int rp=0; rp<NRPotsMAX;...
+    }    // end of for(int stn=0; stn<
+  }      // end of for(int arm=0; arm<2;...
+
+  return;
+}
+
+//-------------------------------------------------------------------------------
+
+void CTPPSRandomDQMSource::analyze(edm::Event const &event, edm::EventSetup const &eventSetup) {
+  Handle<DetSetVector<CTPPSPixelDigi>> pixDigi;
+  event.getByToken(tokenDigi, pixDigi);
+
+  cout << "Before checking digi" << endl;
+  if (!pixDigi.isValid())
+    return;
+  cout << "After checking digi" << endl;
+
+  hBX->Fill(event.bunchCrossing());
+
+  for (int arm = 0; arm < 2; arm++) {
+    for (int stn = 0; stn < NStationMAX; stn++) {
+      if (!StationStatus[stn])
+        continue;
+      for (int rp = 0; rp < NRPotsMAX; rp++) {
+        if (!RPstatus[stn][rp])
+          continue;
+        int index = getRPindex(arm, stn, rp);
+        if (RPindexValid[index] == 0)
+          continue;
+
+        for (int p = 0; p < NplaneMAX; p++) {
+          CTPPSPixelDetId planeId(arm, stn, rp, p);
+          if ((*pixDigi).find(planeId.rawId()) != (*pixDigi).end()) {
+            int n_digis = (*pixDigi)[planeId.rawId()].size();
```

Comment:
The second lookup could be avoided with something like this:
```suggestion
          auto pix_d = pixDigi->find(planeId.rawId());
          if (pix_d != pixDigi->end()) {
            int n_digis = pix_d->size();
```

Reply:
Indeed it's nicer. I'm adding this, thanks!